### PR TITLE
feat(package-operator): add package versioning

### LINKED ISSUE #198: Add Repository support for multiple versions of a package
See https://github.com/glasskube/glasskube/blob/9a6dbccb6c19f205767dfd395decd658b71c4daa/website/docs/03_components/02_package-operator.md for details.

The operator should be able to either install a specific version of a package, or determine the latest version and install that.

### LINKED ISSUE #199: Implement enabling and disabling automatic updates
Automatic updates can be enabled per package by specifying an empty `Version`.

### PR DESCRIPTION
**Pull request by kosmoz**
Closes #198 
Closes #199  

## 📑 Description
Packages can now be created for a specific version of a manifest. 

## ✅ Checks
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required

## ℹ Additional Information
Related PR for docs: #218
Related PR for packages: https://github.com/glasskube/packages/pull/28

### CODE REVIEW

File: api/v1alpha1/package_types.go [hunk 1]
```diff
@@ -39,13 +41,17 @@ type PackageSpec struct {
 
 // PackageStatus defines the observed state of Package
 type PackageStatus struct {
-	Conditions     []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
-	OwnedResources []OwnedResourceRef `json:"ownedResources,omitempty"`
+	Version           string             `json:"version,omitempty"`
```
Comment (pmig) [inline api/v1alpha1/package_types.go:44]: As Helm uses chart and application versions we should also always display both versions and might display the app version more prominent than the chart version.

Comment (kosmoz) [reply api/v1alpha1/package_types.go:44]: This is the version of the Glasskube package. It doesn't have to be neither the chart nor app version, though it *can* be either

**Review by christophenne — CHANGES_REQUESTED** [review]
:rocket: really nice, working perfectly fine. I only have minor naming things to comment

File: internal/controller/package_controller.go [hunk 2]
```diff
@@ -132,13 +126,61 @@ func (r *PackageReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
...
+		}
+	}
+
+	changed := addOwnedResourceRef(&pkg.Status.OwnedPackageInfos, &packageInfo, &packageInfo)
+	return &packageInfo, changed, nil
+}
+
+func generatePackageInfoName(pkg packagesv1alpha1.Package) string {
+	if pkg.Spec.PackageInfo.Version != "" {
+		return escapeResourceName(fmt.Sprintf("%v--%v", pkg.Spec.PackageInfo.Name, pkg.Spec.PackageInfo.Version))
```
[the remark this answers is not in the document]
Comment (kosmoz) [reply internal/controller/package_controller.go:171]: Yes, unfortunately, the naming restrictions are quite strict

File: internal/controller/packageinfo_controller.go [hunk 3]
```diff
@@ -97,8 +102,20 @@ func (r *PackageInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return requeue.Always(ctx, nil)
 }
 
+func (r *PackageInfoReconciler) isLatestVersion(ctx context.Context, packageInfo *packagesv1alpha1.PackageInfo) (bool, error) {
```
[the remark this answers is not in the document]
Comment (kosmoz) [reply internal/controller/packageinfo_controller.go:105]: Good thinking… :+1:

File: pkg/describe/describe.go [hunk 4]
```diff
@@ -21,20 +21,21 @@ func DescribePackage(
 		return nil, nil, nil, err
 	}
 	var status *client.PackageStatus
-	var manifest *v1alpha1.PackageManifest
+	var manifest v1alpha1.PackageManifest
```
[the remark this answers is not in the document]
Comment (kosmoz) [reply pkg/describe/describe.go:24]: Good idea. I changed manifest2 to manifest though, since that's the default import

**Review by christophenne — APPROVED** [review]

**Pull request merged**